frontend/src/pages/user/callback.tsx

```diff
@@ -33,21 +30,28 @@ const Callback = () => {
         code_verifier: verifier,
       }),
     });
-
     return (await resp.json()) as Token;
   };
 
-  const { isPending, error, data } = useQuery({
+  const { isPending, error, data, refetch } = useQuery({
     queryKey: ['token'],
     queryFn: getToken,
+    enabled: false,
   });
 
+  if (!code || !state || !verifier) {
+    console.log('Missing code, state or verifier');
+    return <Navigate to='/login' replace={true} />;
+  }
+
+  refetch();
   if (isPending) {
     return <>Loading</>;
   }
 
   if (error) {
```

Comment:
```suggestion
  const { isPending, isError, data, refetch } = useQuery({
    queryKey: ['token'],
    queryFn: getToken,
    enabled: false,
  });

  if (!code || !state || !verifier) {
    console.log('Missing code, state or verifier');
    return <Navigate to='/login' replace={true} />;
  }

  refetch();
  if (isPending) {
    return <>Loading</>;
  }

  if (isError) {
```
i think i effed up before